common/src/main/java/com/microsoft/identity/common/internal/ui/embeddedwebview/AADWebviewClient.java

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.internal.ui.embeddedwebview;
+
+import android.annotation.TargetApi;
+import android.content.Context;
+import android.net.Uri;
+import android.os.Build;
+import android.webkit.WebResourceRequest;
+import android.webkit.WebView;
+
+import com.microsoft.identity.common.adal.internal.AuthenticationConstants;
+import com.microsoft.identity.common.internal.providers.microsoft.azureactivedirectory.AzureActiveDirectoryAuthorizationRequest;
+import com.microsoft.identity.common.internal.util.StringUtil;
+
+import java.util.Locale;
+
+
+class AADWebViewClient extends OAuth2WebViewClient {
+
+    AADWebViewClient(final Context context, final String redirectURL, final AzureActiveDirectoryAuthorizationRequest request) {
+        super(context, redirectURL, request);
+    }
+
+    public boolean shouldOverrideUrlLoading(WebView view, String url) {
+        if (StringUtil.isEmpty(url)) {
+            throw new IllegalArgumentException("Empty url.");
+        }
+
+        return handleUrl(view, url);
+    }
+
+    @Override
+    @TargetApi(Build.VERSION_CODES.LOLLIPOP)
+    public boolean shouldOverrideUrlLoading(WebView view, WebResourceRequest request) {
+        final Uri requestUrl = request.getUrl();
+        return  handleUrl(view, requestUrl.toString());
+    }
+
+    private boolean handleUrl(final WebView view, final String url) {

```

Comment:
Can we come up with a more descriptive name of what this method does? `handleUrl()` returns boolean doesn't help me understand what this method does at a glance